addons/google_calendar/models/google_calendar.py

```diff
@@ -253,58 +249,60 @@ def generate_data(self, cr, uid, event, isCreating=False, context=None):
         if not event.active:
             data["state"] = "cancelled"
 
-        if not self.get_need_synchro_attendee(cr, uid, context=context):
+        if not self.get_need_synchro_attendee():
             data.pop("attendees")
         return data
 
-    def create_an_event(self, cr, uid, event, context=None):
-        gs_pool = self.pool['google.service']
-        data = self.generate_data(cr, uid, event, isCreating=True, context=context)
+    def create_an_event(self, event):
```

Comment:
I think method is meant for internal use(I don't think so it needs decoration)
I didn't get this statement: you missed to replace "context" with "self.env.context" at couple of places above.